ompi/communicator/ft/comm_ft.c

```diff
@@ -488,8 +489,8 @@ int ompi_comm_ishrink_internal(ompi_communicator_t* comm, ompi_communicator_t**
                                     subreq,
                                     comm->c_coll->coll_iagree_module );
     if( OMPI_SUCCESS != rc ) {
-        ompi_comm_request_return(request);
         OBJ_RELEASE(context->failed_group);
+        ompi_comm_request_return(request);
```

Comment:
I don't see any reason for this change. The request and context can be independently freed, what is the need to change the original order ?

Reply:
This file `comm_ft.c` calls `ompi_comm_request_return()` in 11 places (including the missing one you qualified as "good catch" below). In 9 of these places, the call happens with some `OB_RELEASE()` calls nearby. In most of these instances, `OBJ_RELEASE()` is used *before*  calling `ompi_comm_request_return()`.

Therefore, what I'm simply doing here is to change the call order to be the same everywhere for the sake of coding consistency. In my admittedly subjective view, after all the time I spent in this PR (I'm not acquainted with the ompi code base), consistent coding patterns make things way easier to follow.

This change was indeed technically unneeded a purely aesthetic. If you want me to reverse it, I'll comply, but I would like to very much advocate for keeping it.

Reply:
OK, almost two months after, I do not remember exactly, so what probably happened is that that first instance made me incrementally change things elsewhere. I'll revert these changes and update the first one to keep it consistent.